pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -509,6 +509,189 @@ def pow(self, z):
         return [PauliRot(self.data[0] * z, self.hyperparameters["pauli_word"], wires=self.wires)]
 
 
+class PCPhase(Operation):
+    r"""
+    A projector-controlled phase gate.
+
+    .. math:: \Pi_{\phi}(\phi, dim=2, wires=[0,1]) = \begin{bmatrix}
+                e^{i\phi} & 0 & 0 & 0 \\
+                0 & e^{i\phi} & 0 & 0 \\
+                0 & 0 & 1 & 0 \\
+                0 & 0 & 0 & 1
+            \end{bmatrix}.
+
+    **Details:**
+
+    * Number of wires: AnyWires
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        dim (int): the qubit subspace on which we apply the phase shift
+        wires (Iterable[int, str], Wires): the wire the operation acts on
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+
+    **Example:**
+
+    >>> pc_op = qml.PCPhase(1.23, dim=3, wires=[1,2])
+    >>> qml.matrix(pc_op)
+    array([[0.33423773+0.9424888j, 0.        +0.j       , 0.        +0.j       , 0.        +0.j       ],
+           [0.        +0.j       , 0.33423773+0.9424888j, 0.        +0.j       , 0.        +0.j       ],
+           [0.        +0.j       , 0.        +0.j       , 0.33423773+0.9424888j, 0.        +0.j       ],
+           [0.        +0.j       , 0.        +0.j       , 0.        +0.j       , 1.        +0.j       ]])
+    """
+    num_wires = AnyWires
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+    ndim_params = (0,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    basis = "Z"
+    grad_method = "A"
+    parameter_frequencies = [(1,)]
+
+    def generator(self):
+        r"""The hermitian matrix, which when exponentiated and scaled (i :math:`\phi`), produces
+        the PCPhase unitary."""
+        dim, shape = self.hyperparameters["dimension"]
+        mat = np.diag([1 if index < dim else 0 for index in range(shape)])
+        return qml.Hermitian(mat, wires=self.wires)
+
+    def __init__(self, phi, dim, wires, do_queue=True, id=None):
+        wires = wires if isinstance(wires, Wires) else Wires(wires)
+
+        if not (isinstance(dim, int) and (dim <= 2 ** len(wires))):
+            raise ValueError(
+                f"The projected dimension {dim} must be an integer that is less then or equal to "
+                f"the max size of the matrix {2 ** len(wires)}. Try adding more wires."
+            )
+
+        super().__init__(phi, wires=wires, do_queue=do_queue, id=id)
+        self.hyperparameters["dimension"] = (dim, 2 ** len(wires))
```

Comment:
This quantity is needed in the static method `compute_matrix` which has no idea about the operator's wires attribute since it is a static method. Adding it here reduces lines of code.